src/compliance/ec2_compliance.py

```diff
@@ -16,35 +16,71 @@ def analyse(self, audit: Audit) -> Set[Findings]:
+                    if launch_time_days_ago > MAX_AMI_AGE_DAYS:
+                        findings.add(
+                            self.create_ami_age_finding(
+                                account=account,
+                                instance=instance,
+                                findings={
+                                    f"Instance:  `{instance['id']}`",
+                                    f"Component: `{instance['component']}`",
+                                    f"Started:   `{launch_time_days_ago}` days ago.",
+                                },
+                                description=f"Running image AMI has been deleted"
+                                f" but instance has been running for over {MAX_AMI_AGE_DAYS} days!"
+                                " The older an instance becomes the higher the likelihood"
+                                " the image is missing important security patches.",
+                            )
+                        )
+                else:
+                    try:
+                        image_creation_date = datetime.fromisoformat(image_creation_date)
+                    except ValueError:
+                        self.log_date_parse_error(instance["id"], instance["image_creation_date"])
+                        continue
+
+                    image_creation_days_ago = (datetime.now(timezone.utc) - image_creation_date).days
+                    if image_creation_days_ago > MAX_AMI_AGE_DAYS:
+                        findings.add(
+                            self.create_ami_age_finding(
+                                account=account,
+                                instance=instance,
+                                findings={
+                                    f"Instance:  `{instance['id']}`",
+                                    f"Component: `{instance['component']}`",
+                                    f"Created:   `{image_creation_days_ago}` days ago.",
+                                },
+                                description=f"Running image AMI created over {MAX_AMI_AGE_DAYS} days ago."
+                                " The older an AMI becomes the higher the likelihood"
+                                " the image is missing important security patches.",
+                            )
                         )
-                    )
 
         return findings
+
+    @staticmethod
+    def create_ami_age_finding(
+        account: Dict[str, Any], instance: Dict[str, Any], findings: set[str], description: str
+    ) -> Findings:
+        return Findings(
+            compliance_item_type="ami_creation_age",
+            account=Account.from_dict(account["account"]),
+            item=instance["component"],
+            findings=findings,
+            description=description,
+        )
+
+    def log_date_parse_error(self, instance_id: str, date_string: str) -> None:
+        self.logger.error(
+            f"image_creation_date '{date_string}'"
```

Comment:
this could be  instance["launch_time"] as well, as it's called for both

Reply:
Can ether be launch_time or image_creation_date will fix the error message to be aware of that